.azure-pipelines/markers_check/markers_check.py

```diff
@@ -0,0 +1,56 @@
+import re
+import os
+import sys
+import logging
+
+from natsort import natsorted
+
+
+def collect_all_scripts():
+    '''
+    This function collects all test scripts under the folder 'tests/'
+    and get the topology type marked in the script
+    '''
+    location = sys.argv[1]
+
+    # Recursively find all files starting with "test_" and ending with ".py"
+    # Note: The full path and name of files are stored in a list named "files"
+    files = []
+    for root, dirs, file in os.walk(location):
+        for f in file:
+            if f.startswith("test_") and f.endswith(".py"):
+                files.append(os.path.join(root, f))
+    files = natsorted(files)
+
+    # Open each file and search for regex pattern
+    pattern = re.compile(r"[^@]pytest\.mark\.topology\(([^\)]*)\)")
+
+    for f in files:
+        has_markers = False
+        # Remove prefix from file name:
+        filename = f[len(location) + 1:]
+        try:
+            with open(f, 'r') as file:
```

Comment:
Same potential conflict between variable name `file` and builtin module `file`.

Reply:
Modified to script.